src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3583,10 +3583,11 @@ async def __common_validation(self, verrors, schema, data, mode, parent=None, cu
 
                 if 'volblocksize' in data:
 
-                    if data['volblocksize'].isdigit():
-                        block_size = int(data['volblocksize'])
+                    blocksize = f'{data["volblocksize"]}B' if data['volblocksize'].isdigit() else data['volblocksize']
```

Comment:
Does it make more sense to do something like this? It removes a few if/else branches and an `int` cast. What do you think?
```suggestion
                    if data['volblocksize'][:3] == '512':
                        block_size = 512
                    else:
                        block_size = int(blocksize[:-1]) * 1024
```